### PR TITLE
PROF-10079: jvmti based wallclock sampler

### PR DESCRIPTION
**Pull request by richardstartin**
**What does this PR do?**:
Introduces a new wallclock sampler not based on signals, which uses JVMTI APIs instead.

**Motivation**:

**Additional Notes**:

**How to test the change?**:

**For Datadog employees**:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles
  credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddprof-lib/src/main/cpp/wallClock_jvmti.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    javaThreads.reserve(_reservoir_size);
+    ReservoirSampler<int> reservoir(_reservoir_size);
+    int self = OS::threadId();
+    u64 counter = 0;
+
+
+    while (_running.load(std::memory_order_relaxed)) {
+        int numFailures = 0;
+        int threadsAlreadyExited = 0;
+        std::vector<int> sample = reservoir.sample(tids);
```
Comment (jbachorik) [inline ddprof-lib/src/main/cpp/wallClock_jvmti.cpp:85]: Where is the `tids` vector filled with the TIDs?
I might be blind but I can't seem to find it :(

Comment (jbachorik) [reply ddprof-lib/src/main/cpp/wallClock_jvmti.cpp:85]: In the ASGCT version I see the thread_filter taking care of that - https://github.com/DataDog/java-profiler/pull/118/files#diff-a368edb3b62aef3c92f4e699fc69e1a6fabed5d158b808dabda9af2018073ddaR158-R170

Comment (richardstartin) [reply ddprof-lib/src/main/cpp/wallClock_jvmti.cpp:85]: good catch, the test wasn't testing what I thought it was. I've added some extra tests to verify that the profiler under test is the one in use.

**Comment by r1viollet** [comment]
I will convert to draft to ensure we can keep this around for documentation, though we are not planning to integrate this as of now.

**Comment by jbachorik** [comment]
@r1viollet I hope we will get this finished. The ASGCT and I/O interrupted by signals issues would be great to avoid...

**Pull request closed without merging**